core/trino-main/src/main/java/io/trino/execution/scheduler/ResultsConsumer.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.scheduler;
+
+import io.trino.execution.RemoteTask;
+import io.trino.sql.planner.plan.PlanFragmentId;
+
+public interface ResultsConsumer
```

Comment:
How about `SourceTaskLifecycleListener` or something that conveys it's a listener for events related to source task creation?